src/main/java/org/embulk/output/sftp/SftpFileOutput.java

```diff
@@ -211,27 +202,68 @@ public TaskReport commit()
 
     private void closeCurrentFile()
     {
-        if (currentFile == null) {
-            return;
+        if (localOutput != null) {
+            try {
+                localOutput.close();
+            }
+            catch (IOException ex) {
+                throw Throwables.propagate(ex);
+            }
         }
+    }
 
+    private Void uploadFile(final String remotePath)
+    {
         try {
-            currentFileOutputStream.close();
-        }
-        catch (IOException e) {
-            logger.info(e.getMessage());
+            return retryExecutor()
+                    .withRetryLimit(maxConnectionRetry)
+                    .withInitialRetryWait(500)
+                    .withMaxRetryWait(30 * 1000)
+                    .runInterruptible(new Retryable<Void>() {
+                        @Override
+                        public Void call() throws InterruptedException, IOException, RetryGiveupException
+                        {
+                            FileObject remoteFile = newSftpFile(getSftpFileUri(remotePath));
```

Comment:
It seems that `newSftpFile` method also has retry. Do you have the special reason why a retry is nested?